### PR TITLE
Fix flaky tests

### PR DESCRIPTION
**Pull request by zucchini-nlp**
# What does this PR do?

Fixes flakiness in some generation tests for VLMs. We have to pass the `bad_ids` in all tests that call generate with tiny models.

Plus for Mllama we don't have to check the generation stage that way anymore, since we decided not to support muti-turn conversations.

### CODE REVIEW

**Review by gante — COMMENTED** [review]
I don't like this approach 😅 

The root issue is in the models, which should never generate certain tokens (as opposed to being in a poorly parameterized test). Merging this PR makes the test pass, but not only doesn't fix the issue, but also makes testing more complex. The problem will just become bigger as we add more VLMs.

Given that `@is_flaky()` also fixes the failing test, but adds a bigger sense of urgency (we know that we shouldn't have this tag in the long run) and doesn't make tests more complex, I'm heavily biased towards it.

The exception would be if ALL tests have the same problem. In that case, I'd agree merging tests as is, but we would have to put the underlying fix at a very high priority.

**Comment by zucchini-nlp** [comment]
I see, this is not a very nice workaround to force certain tokens from being generated. Actually the test for mllama would not be flaky just by changing the check in prepare-inputs, as I did in this PR

We can get rid of flakiness for now, but take a note for future that fixing it in a proper way is very important. Interestingly passing logits processot makes other tests flaky.... So yeah, that wouldn't work

**Review by gante — APPROVED** [review]
LGTM is this fixes the issue :)

**Comment by zucchini-nlp** [comment]
@ArthurZucker can you merge pls? The tests is failing because we have unused attribute, but we can't remove it for BC. Also I want all VLMs have such an attr by default, like `bos_token` etc.

For the default one, I'll open another PR to see if everyone agrees

**Comment by gante** [comment]
@zucchini-nlp we can't force-merge, this failing check would start showing up in all PRs :P In this case we have to fix the underlying issue or add an exception to the script that checks it :) (lmk if you'd like pointers)

**Comment by zucchini-nlp** [comment]
@gante okey, let's do the second option as part of this PR then :)

Added image token id as always allowed, so that all VLMs can have it even if not used directly in modeling. I believe this is useful already in tests + may be useful in other cases in the future

**Comment by gante** [comment]
@LysandreJik FYI -- merged this PR, as it is blocking other PRs :)

**Comment by LysandreJik** [comment]
Thanks for doing so!
